### PR TITLE
ETRAM - 7

### PR DESCRIPTION
**Pull request by mradagonzalez**
Added working functionality to created endpoints:
- Article: 
- - Get child articles of a parent page
- - Get content from a page
- - Download images from an article
- Project
- - Get top level articles from a project
- - Get table of contents from a project
- Added postman project with working examples (dev env)

### CODE REVIEW

File: src/main/java/com/ixxus/etram/experttrack/infrastructure/rest/ArticleController.java [hunk 1]
```diff
@@ -15,23 +25,26 @@ public class ArticleController {
 
     //TODO check if projectId is needed in path
```
Comment (eriveraCCC) [inline src/main/java/com/ixxus/etram/experttrack/infrastructure/rest/ArticleController.java:26]: delete comment

File: src/main/java/com/ixxus/etram/experttrack/model/entity/article/CustomArticleHtmlContentEntity.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Entity
+@NoArgsConstructor
+public class CustomArticleHtmlContentEntity {
```
Comment (eriveraCCC) [inline src/main/java/com/ixxus/etram/experttrack/model/entity/article/CustomArticleHtmlContentEntity.java:15]: does this entity needs to reference a table?

File: src/main/java/com/ixxus/etram/experttrack/model/entity/article/CustomArticleTopLevelEntity.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Entity
+@NoArgsConstructor
+public class CustomArticleTopLevelEntity {
```
Comment (eriveraCCC) [inline src/main/java/com/ixxus/etram/experttrack/model/entity/article/CustomArticleTopLevelEntity.java:15]: does this entity needs to reference a table?

File: src/main/java/com/ixxus/etram/experttrack/model/entity/project/CustomProjectTocEntity.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Entity
+@NoArgsConstructor
+public class CustomProjectTocEntity {
```
Comment (eriveraCCC) [inline src/main/java/com/ixxus/etram/experttrack/model/entity/project/CustomProjectTocEntity.java:15]: does this entity needs to reference a table?

**Review by eriveraCCC — COMMENTED** [review]
Review comments

File: src/main/java/com/ixxus/etram/experttrack/infrastructure/rest/ArticleController.java (hunk 1 above)
Comment (mradagonzalez) [reply src/main/java/com/ixxus/etram/experttrack/infrastructure/rest/ArticleController.java:26]: Done

File: src/main/java/com/ixxus/etram/experttrack/model/entity/article/CustomArticleChildEntity.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Entity
+@NoArgsConstructor
+public class CustomArticleChildEntity {
+
```
[the remark this answers is not in the document]
Comment (mradagonzalez) [reply src/main/java/com/ixxus/etram/experttrack/model/entity/article/CustomArticleChildEntity.java:16]: The entity doesn't reference a table, it references a result map, so it doesn't need the "table" parameter

**Pull request merged**